cmake/modules/hunter_pkgconfig_export_target.cmake

```diff
@@ -0,0 +1,128 @@
+# ----------------------------------------------------------------------
+# Auto creation of CMake targets from pkg-config
+#
+# Copyright (C) 2017 Alexandre Pretyman. All rights reserved.
+#
+# Looks for a pkgconfig module of a given name and exports a cmake
+# target of the same name
+# ----------------------------------------------------------------------
+
+include(FindPkgConfig)
+
+include(hunter_internal_error)
+include(hunter_status_debug)
+
+function(hunter_pkgconfig_export_target PKG_CONFIG_MODULE)
+  if(TARGET "${PKG_CONFIG_MODULE}")
+    return()
+  endif()
+  pkg_check_modules(${PKG_CONFIG_MODULE} ${PKG_CONFIG_MODULE})
```

Comment:
I just debugged CMake's version of the `IMPORTED_TARGET` and yes, it is bugged.

It takes the flags and tries to convert them into full paths to the files with `find_library`.

pkg-config LDFLAGS start with: `-L<root-id>/lib -lx264 -lpthread -lm -ldl`.
CMake's IMPORTED target takes the first argument `<root-id>/lib` and uses as a directory to call `find_library(...)` on the libraries `x264`, `pthread`, `m`, `dl`. The result will be added to the imported target's libraries. However it finds only `x264`, but not others which are present in the toolchain, they become `pthread-NOTFOUND`, `m-NOTFOUND`, `dl-NOTFOUND`. it it append these values to a var `_libs`, and below checks `if(_libs) add libs to target`, where this if returns false because of the `NOTFOUND` elements (last element?). So it does not add any libs to the target.

Possible fix: if `find_library(...)` returns `lib-NOTFOUND` then use the given pkg-config's linking flags `-lpthread`, etc.. instead of trying to link to the absolute path of the lib.

Hope this explanation was clear :-)